### ISSUE TITLE
False positives on ThreadLocal get and set

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#8944]
I'm getting false positives on `UnusedPrivateProperty` and `ExplicitCollectionElementAccessMethod` in an actual class.

## Expected Behavior

Not to have the violation.

## Observed Behavior

There is a violation.

## Steps to Reproduce

```kotlin
internal actual class TransactionsThreadLocal actual constructor() {
  private val transactions = ThreadLocal<Transacter.Transaction>()

  internal actual fun get(): Transacter.Transaction? = transactions.get()

  internal actual fun set(transaction: Transacter.Transaction?) {
    transactions.set(transaction)
  }
}
```

## Your Environment

* Version of detekt used: 2.0.0-alpha.1

**Comment by eygraber** [comment]
Happening [here](https://github.com/eygraber/sqldelight-androidx-driver/blob/master/library/src/jvmMain/kotlin/com/eygraber/sqldelight/androidx/driver/TransactionsThreadLocal.kt)

**Comment by atulgpt** [comment]
Is this also happening without the use `actual` in the class?

**Comment by eygraber** [comment]
I didn't test that.

**Comment by atulgpt** [comment]
I tested the TC

```kotlin
@Test
fun `threadlocal get and set`() {
    val code = """
        internal class TransactionsThreadLocal constructor() {
            private val transactions = ThreadLocal<Transaction>()

            internal fun get(): Transaction? = transactions.get()

            internal fun set(transaction: Transaction?) {
                transactions.set(transaction)
            }
        }

        internal class Transaction
    """.trimIndent()
    assertThat(subject.lintWithContext(env, code)).isEmpty()
}
```

and the above TC is passing. This seems to be dupe of https://github.com/detekt/detekt/issues/8641
